web/readers.go

```diff
@@ -206,11 +211,17 @@ func handlerHypha(w http.ResponseWriter, rq *http.Request) {
 			contents = mycoopts.Media(h, lc) + contents
 		}
 
+		cats := []string{}
+		for _, category := range categories.CategoriesWithHypha(h.CanonicalName()) {
+		    cats = append(cats, "cat-" + category)
+		}
```

Comment:
That would not work. According to the [HTML spec](https://html.spec.whatwg.org/multipage/syntax.html#attributes-2):

> Attributes have a name and a value. Attribute names must consist of one or more characters other than [controls](https://infra.spec.whatwg.org/#control), U+0020 SPACE, U+0022 ("), U+0027 ('), U+003E (>), **U+002F (/)**, U+003D (=), and [noncharacters](https://infra.spec.whatwg.org/#noncharacter). In the HTML syntax, attribute names, even those for [foreign elements](https://html.spec.whatwg.org/multipage/syntax.html#foreign-elements), may be written with any mix of [ASCII lower](https://infra.spec.whatwg.org/#ascii-lower-alpha) and [ASCII upper alphas](https://infra.spec.whatwg.org/#ascii-upper-alpha).

Meanwhile, the slash / is a valid character in canonical hypha names.

So, the actual hypha name has to be in attribute's value. You can't have multiple attributes with the same name, so they'll have to be in one attribute, separated by some character. `:` would work fine.

See [docs](https://mycorrhiza.wiki/help/en/hypha#Hypha_names) for the list of disallowed characters.

Reply:
That makes sense. In that case I think the `:` seperator should be the starting and ending character as well (`data-rrh-cats=":foo:bar:"`) to make the css selections easier.